doc/visual-programming/source/widgets/data/mergedata.md

```diff
@@ -12,24 +12,79 @@ Merges two datasets, based on values of selected attributes.
 
 - Data: dataset with features added from extra data
 
-The **Merge Data** widget is used to horizontally merge two datasets, based on values of selected attributes. In the input, two datasets are required, data and extra data. The widget allows selection of an attribute from each domain, which will be used to perform the merging. The widget produces one output. It corresponds to instances from the input data to which attributes from input extra data are appended.
+The **Merge Data** widget is used to horizontally merge two datasets, based on the values of selected attributes (columns). In the input, two datasets are required, data and extra data. The widget allows selection of one or more attributes from each domain, which will be used to perform the merging. The widget produces one output. It corresponds to the instances from the input data to which attributes (columns) from input extra data are appended.
 
-Merging is done by values of selected (merging) attributes. First, the value of the merging attribute from Data is taken and instances from Extra Data are searched for matching values. If more than a single instance from Extra Data was to be found, the attribute is removed from available merging attributes.
+Merging is done by values of selected (merging) attributes. First, the value of the merging attribute from Data is taken and instances from Extra Data are searched for matching values. If the selected attribute does not contain unique values (in other words, the attribute has duplicate values), the attribute is removed from the available merging attributes.
 
-![](images/MergeData-stamped.png)
+Merge Data can merge also on more than one attribute. Click on the plus icon to add the attribute to merge on. The final result have to be unique combinations for each individual row.
 
-1. Information on Data
-2. Information on Extra Data
-3. Merging type.
-   **Append columns from Extra Data** outputs all instances from Data appended by matching instances from Extra Data. When no match is found,unknown values are appended.
-   **Find matching rows** outputs only matching instances.
-   **Concatenate tables, merge rows** outputs all instances from both inputs, even though the match may not be found. In that case unknown values are assigned.
-4. List of comparable attributes from Data
-5. List of comparable attributes from Extra Data
+![](images/Merge-Data-stamped.png)
+
+1. Information on main data.
+2. Information on data to append.
+3. Merging type:
+   - **Append columns from Extra Data** outputs all instances from Data appended by matching instances from Extra Data. When no match is found,unknown values are appended.
+   - **Find matching pairs of rows** outputs only matching instances.
+   - **Concatenate tables** outputs all instances from both inputs, even though the match may not be found. In that case unknown values are assigned.
```

Comment:
- **Concatenate tables** treats both data sources symmetrically. The output is similar as for the first option, except that non-matched values from Extra data are appended at the end.